### PR TITLE
cleans up suits/miscellaneous.dm

### LINKED ISSUE #8497: Raincoat makes you naked
(WEB REPORT BY: shiggypiggy REMOTE: 158.69.55.2:7777)

> Revision (Should be above if you're viewing this from ingame!)
> f7fb8a3fa157d03616c37d39c1cc00b832e12b7a 
> General description of the issue
> While wearing the raincoat, you look naked under it
> 
> What you expected to happen
> Not be naked
> 
> What actually happened
> I was naked
> 
> Steps to reproduce if possible
> Get a raincoat and put it on, enjoy exhibitionism

### LINKED ISSUE #8595: Liberty coat makes clothes under it invisible
(WEB REPORT BY: cowboybuttsdrivemenuts REMOTE: 158.69.55.2:7777)
Revision: 255da17316da9aa055a16160c84e1aa831873695 

> As in the title, putting on the liberty coat into your exosuit slot overrides your jumpsuit and makes it appear that you're in your undies
> 
> I expected to be able to put on a liberty coat and wear other clothes under it
> 
> What actually happened
> The clothes are invisible
> 
> Steps to reproduce if possible
> Put on a liberty coat from the autodrobe.

### LINKED ISSUE #9063: Fairy wings make your jumpsuit not display.
(WEB REPORT BY: konater REMOTE: 158.69.55.2:7777)
Sorry I don't know the flag or whatever. Putting fairy wings on reveal your underwear to all despite having a jumpsuit on.

### PR DESCRIPTION
**Pull request by Intigracy**
fix #8497
fix #9063
fix #8595

removes item_state = from items that don't have an inhand icon by that name

Yes I tested this.

Unrelated to PR:
TIL that clothing of type /under needs the _color var (which washing machines added/use to change the color of jumpsuits) to determine it's onmob overlay icon_state, since just having an icon_state doesn't work.

TL;DR fuck oldcode

### CODE REVIEW

File: code/modules/clothing/suits/miscellaneous.dm [hunk 1]
```diff
@@ -211,51 +206,32 @@
 		return
 	usr.update_inv_wear_suit()
 
-//pyjamas
```
Comment (Intigracy) [inline code/modules/clothing/suits/miscellaneous.dm:214]: removed because duplicate defines, these two are unused and broken, the other pajamas are correctly in clothing/under/misc

**Comment by gbasood** [comment]
don't even know what to tag this

**Comment by Intigracy** [comment]
Well it is a bugfix really, since it's just changing the flags so clothes don't make you naked.

The rest is just cleanup.
